### PR TITLE
add parameter for exception ignore in kafka2

### PR DESCRIPTION
**Pull request by JiHyunSong**
hello, 
for using in output kafka2, I made it to skip ruby-kafka exception. (like skip Kafka::MessageSizeTooLarge)
and add some parameter to backup what chunks make exception occur.

I'm not used to ruby, so give me some feedbacks ^^;

thanks

### CODE REVIEW

**Comment by soultemple** [comment]
👍 We need it to handle Kafka::MessageSizeTooLarge.

File: lib/fluent/plugin/out_kafka2.rb [hunk 1]
```diff
@@ -36,6 +36,9 @@ class Fluent::Kafka2Output < Output
 
     config_param :get_kafka_client_log, :bool, :default => false
 
+    config_param :ignore_exceptions, :string, :default => nil, :desc => "Ignore exception list split by comma (,)"
```
Comment (repeatedly) [inline lib/fluent/plugin/out_kafka2.rb]: use `array` type, not `string`: https://docs.fluentd.org/v1.0/articles/api-config-types#:array

File: lib/fluent/plugin/out_kafka2.rb [hunk 2]
```diff
@@ -218,12 +222,22 @@ def write(chunk)
         end
       end
     rescue Exception => e
+      ignore = false
+      if ignore_exceptions.size > 0
```
Comment (repeatedly) [inline lib/fluent/plugin/out_kafka2.rb]: Use `Array#include?`.

File: lib/fluent/plugin/out_kafka2.rb [hunk 3]
```diff
@@ -218,12 +222,22 @@ def write(chunk)
...
+        }
+      end
+
       log.warn "Send exception occurred: #{e}"
       log.warn "Exception Backtrace : #{e.backtrace.join("\n")}"
+      log.warn "Exception ignored in tag : #{tag}" if ignore
       # For safety, refresh client and its producers
       refresh_client(false)
+      # raise UnrecoverableError for backup ignored exception chunk
+      raise Fluent::UnrecoverableError if exception_backup
```
Comment (repeatedly) [inline lib/fluent/plugin/out_kafka2.rb]: `if ignore && exception_backup`?
Recoverable errors are also ignored?

Comment (JiHyunSong) [reply lib/fluent/plugin/out_kafka2.rb]: `if ignore && exception_backup` (even if Fluent::UnrecoverableError defined in conf) Recoverable error will not ignored.

what do you think about which is the best way to backup chunk? (I want to know which log makes exception)

**Comment by JiHyunSong** [comment]
@repeatedly check it again, please ... :D

**Comment by repeatedly** [comment]
Thanks for the patch!

**Pull request merged**